core/src/utils/forms/compare-with-utils.ts

```diff
@@ -0,0 +1,44 @@
+type CompareFn = (currentValue: any, compareValue: any) => boolean;
+
+/**
+ * Uses the compareWith param to compare two values to determine if they are equal.
+ *
+ * @param currentValue The current value of the control.
+ * @param compareValue The value to compare against.
+ * @param compareWith The function or property name to use to compare values.
+ */
+export const compareOptions = (
+  currentValue: any,
+  compareValue: any,
+  compareWith?: string | CompareFn | null
+): boolean => {
+  if (typeof compareWith === 'function') {
+    return compareWith(currentValue, compareValue);
+  } else if (typeof compareWith === 'string') {
+    return currentValue[compareWith] === compareValue[compareWith];
```

Comment:
`currentValue` can be `null` with Angular forms, so this throws errors when using the property string variant :(

Reply:
@dietergeerts Can you file an issue and include a reproduction? The team doesn't typically monitor closed PRs.